packages/common/client-utils/README.md

```diff
@@ -0,0 +1,69 @@
+# @fluid-internal/client-utils
+
+This package is intended for sharing and promoting utility functions across packages in the Fluid Framework repo.
+
+Use outside of the Fluid Framework repo is not supported or recommended.
+
+## Adding code to this package
+
+As a utility package, this package does not have a strong identity. This means that it's easy to become a "dumping
+ground" for code that we think we should share but doesn't have an obvious home. We try to avoid dumping things into
+utility packages, and this one is no exception.
+
+New code should only be added to this package in rare circumstances. In most cases, the code would be better placed in a
+package with a clear identity (e.g. an "events" package for shared event infrastructure) or not shared at all.
+
+## Requirements
+
+This package has important requirements for the code within it.
+
+1. Code within this package should require some external dependencies. If it does not, then the **core-utils** package
+   is a better location.
+1. **All exports must be designated `@internal`.** This code is intended for use within the Fluid Framework only.
```

Comment:
Is this necessary? Since we document the entire package as being for internal use only? Is this in preparation for the api-extractor internal type stripping so external customers basically see an empty package?

No argument from me on this, just wondering what the rationale is / what the plans are :)

Reply:
It seemed to be the most accurate to our purposes. If we don't intend the APIs to be used externally, let's mark them as such. Package names help, README notes help, but defense-in-depth. I don't feel too strongly about it though.

I was thinking this might make it easier for partners like Loop to find APIs they shouldn't be using, even though they're not deprecated. But that was more of an idea than a plan.

Reply:
Like I said, no argument from me. An added line of defense sounds good to me, if someone does happen to stumble onto one of our internal packages 🙂